Fixes for building on linux

jamesdillonharvey opened the pull request:
Missing comma in LIBS array
Missing include when using non standard libevent

jharvey-fds committed:
Fixes for building on linux, missing comma in LIBS array and missing include when using non standard libevent

fquinner commented:
Looks good - thanks James.

jamesdillonharvey commented:
Hi,

I am building the OMNM payload lib, I cant seem to find the enum entry for
it in your zmq branch of OpenMama and it caused the build to fail.

I have added it locally as follows.

--- a/mama/c_cpp/src/c/mama/msg.h
+++ b/mama/c_cpp/src/c/mama/msg.h
@@ -51,6 +51,7 @@ typedef enum mamaPayloadType_
     MAMA_PAYLOAD_KWANTUM    = 'K',
     MAMA_PAYLOAD_UMS        = 'L',
     MAMA_PAYLOAD_INRUSH     = 'N',
+    MAMA_PAYLOAD_OMNM       = 'O',
     MAMA_PAYLOAD_TICK42RMDS = 'P',
     MAMA_PAYLOAD_QPID       = 'Q',
     MAMA_PAYLOAD_TIBRV      = 'R',

Not sure if that's correct but it seems to build now.

Thanks

James


On Tue, Nov 10, 2015 at 4:32 PM, fquinner <notifications@github.com> wrote:

> Looks good - thanks James.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fquinner/OpenMAMA-zmq/pull/7#issuecomment-155478075>.
>

fquinner commented:
Hi James,

Interesting - i thought i had added that already to the feature-omzmq branch of my fork but it must not have gotten pushed. There's another change needed in msg.c as well - I'll put them into the feature-omzmq branch when i get to my dev laptop.

Cheers,
Frank

jamesdillonharvey commented:
Hi Frank,

Just a quick note to say great job on the zmq bridge and omnm payload.

I have cobbled it into our market data framework and intial tests show a
huge performance increase over the qpid bridge/payload.

I am having a few issues with string fields when the size/data changes but
other than that its just works.

Do you have any plans to add this to the main line opemmama repo?

Thanks

James.

On Thu, Nov 26, 2015 at 8:53 AM, fquinner <notifications@github.com> wrote:

> Hi James,
>
> Interesting - i thought i had added that already to the feature-omzmq
> branch of my fork but it must not have gotten pushed. There's another
> change needed in msg.c as well - I'll put them into the feature-omzmq
> branch when i get to my dev laptop.
>
> Cheers,
> Frank
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fquinner/OpenMAMA-zmq/pull/7#issuecomment-159850638>.
>

fquinner commented:
Thank you and I'm glad to see you are making use of the projects! That variable width field thing looks like a bug (actually had a TODO in the original code to fix that) - just fixed it over in https://github.com/fquinner/OpenMAMA-omnm/issues/1.

I don't have any plans to submit this to the main OpenMAMA repo at the moment because I quite like the idea of having an independent bridge out there and putting the onus on the OpenMAMA project to make this really easy (I'm also involved in the OpenMAMA side so I'm on the hook at that end too).

I've already submitted a few changes and new functionality back to core OpenMAMA to make it easier for projects like this to exist and I plan on submitting a few more once dynamic bridge loading is completed (another reason why I haven't submitted the enum to core OpenMAMA yet - once dynamic loading is working, the enum will be optional).

jamesdillonharvey commented:
Hi,

Yes that all makes sense to me, allowing for dynamic loading sounds better
than bundling it all into the main repo.

I still had some trouble after pulling your fix, my system ran a little
longer but this time crashed at random points with heap corruption.
I think I have a fix. It works with my live system (real market data ~
20,000 msg/s) and I have added a unit test.

https://github.com/fquinner/OpenMAMA-omnm/pull/2

I messed up the initial commit (more used to perforce) and added loads of
extra files (which i removed in the second) but it should be correct now.

James



On Sun, Nov 29, 2015 at 4:10 PM, fquinner <notifications@github.com> wrote:

> Thank you and I'm glad to see you are making use of the projects! That
> variable width field thing looks like a bug (actually had a TODO in the
> original code to fix that) - just fixed it over in
> fquinner/OpenMAMA-omnm#1
> <https://github.com/fquinner/OpenMAMA-omnm/issues/1>.
>
> I don't have any plans to submit this to the main OpenMAMA repo at the
> moment because I quite like the idea of having an independent bridge out
> there and putting the onus on the OpenMAMA project to make this really easy
> (I'm also involved in the OpenMAMA side so I'm on the hook at that end too).
>
> I've already submitted a few changes and new functionality back to core
> OpenMAMA to make it easier for projects like this to exist and I plan on
> submitting a few more once dynamic bridge loading is completed (another
> reason why I haven't submitted the enum to core OpenMAMA yet - once dynamic
> loading is working, the enum will be optional).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fquinner/OpenMAMA-zmq/pull/7#issuecomment-160428117>.
>
